ACS-4966 Add path to Category model

MichalKinas opened the pull request:
https://alfresco.atlassian.net/browse/ACS-4966
This PR is about adding `path` param to Categories API.

tpage-alfresco commented:
Sorry - I added people prematurely to this PR. @MichalKinas You haven't added a description about the new `include=path` option to the category endpoints.

tpage-alfresco commented:
Sorry - I added people prematurely to this PR.

@MichalKinas You haven't added a description about the new `include=path` option to the category endpoints.

MichalKinas commented:
> Sorry - I added people prematurely to this PR. @MichalKinas You haven't added a description about the new `include=path` option to the category endpoints.

Right thank you, added now

azakrzewski-hy commented:
> > Sorry - I added people prematurely to this PR. @MichalKinas You haven't added a description about the new `include=path` option to the category endpoints.
> 
> Right thank you, added now

Should we keep * path field in both descriptions: for categoryEntryIncludeParam and categoryEntryIncludePathParam ?

MichalKinas commented:
> > > Sorry - I added people prematurely to this PR. @MichalKinas You haven't added a description about the new `include=path` option to the category endpoints.
> > 
> > 
> > Right thank you, added now
> 
> Should we keep * path field in both descriptions: for categoryEntryIncludeParam and categoryEntryIncludePathParam ?

Yes `categoryEntryIncludeParam` is used when both `count` and `path` options are available and `categoryEntryIncludePathParam` is used when only `path` option is available for given endpoint

tpage-alfresco commented on src/main/webapp/definitions/alfresco-core.yaml:
```diff
@@ -228,6 +228,18 @@ parameters:
+    items:
+      type: string
+    collectionFormat: csv
+  categoryEntryIncludePathParam:
```
It's a little confusing to have two of these, but we should fix it as part of https://alfresco.atlassian.net/browse/ACS-4923

tpage-alfresco commented:
I think you're still missing some endpoints in this change. I think you added `include=path` to GET category, PUT category, GET subcategories, POST subcategories, GET nodes/{id}/categories and POST nodes/{id}/categories in the other PR.

MichalKinas commented:
> I think you're still missing some endpoints in this change. I think you added `include=path` to GET category, PUT category, GET subcategories, POST subcategories, GET nodes/{id}/categories and POST nodes/{id}/categories in the other PR.

@tpage-alfresco are you sure? I've just double checked and each Category related endpoint except DELETE has `include=path` added in its description.

tpage-alfresco reviewed (approved):
No, you're right - this looks fine. Sorry for the confusion!
